cmake/CMakeLists.txt

```diff
@@ -1519,7 +1519,7 @@ if (onnxruntime_ENABLE_TRAINING)
   list(APPEND onnxruntime_EXTERNAL_LIBRARIES tensorboard)
 endif()
 
-if (UNIX AND onnxruntime_USE_NCCL)
+if (UNIX OR onnxruntime_USE_NCCL)
```

Comment:
If I understand correctly, now all our packages are built without this feature(NCCL).  And this PR will enable this feature in all ONNX Runtime's training packages? We should think it again.

Reply:
No. Read the whole block and you'll see that nothing gets enabled by default. There are two more conditional blocks inside (one of them a repetition of `onnxruntime_USE_NCCL`, which renders it redundant at the top): https://github.com/microsoft/onnxruntime/blob/4c3c809bdbcde4ea96f0a31a242ca6877a10c40a/cmake/CMakeLists.txt#L1532-L1627

Reply:
So the two inner blocks should be
```
if (UNIX AND onnxruntime_USE_MPI) 
```
And 
```
if (UNIX AND onnxruntime_USE_NCCL) 
```